src/errors.rs

```diff
@@ -33,28 +33,14 @@ pub enum InternalError {
     InternalInvariantFailed,
     #[error("Paillier error: `{0}`")]
     PaillierError(#[from] paillier::Error),
```

Comment:
This one also involves a larger code rewrite, and may have similar issues to `CouldNotGenerateProof`